openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -249,6 +291,203 @@ const GlossaryTermTab = ({
     return data;
   }, [glossaryTerms, permissions]);
 
+  const listOfVisibleColumns = useMemo(() => {
+    return ['name', 'description', 'owners', 'status', 'new-term'];
+  }, [permissions]);
+
+  const defaultCheckedList = useMemo(() => {
+    return columns
+      .filter((column) => listOfVisibleColumns.includes(column.key as string))
+      .map((column) => column.key)
+      .filter((key): key is string => key !== undefined);
+  }, [columns, listOfVisibleColumns]);
+
+  const [checkedList, setCheckedList] = useState<string[]>(defaultCheckedList);
+  const [tempCheckedList, setTempCheckedList] = useState<string[]>([
+    ...checkedList,
+  ]);
+
+  const [isDropdownVisible, setIsDropdownVisible] = useState<boolean>(false);
+
+  const [options, setOptions] = useState<{ value: string; label: string }[]>(
+    columns
+      .map(({ key, title }) => ({
+        label: title as string,
+        value: key as string,
+      }))
+      .filter((col) => col.value !== 'name')
+  );
+
+  const newColumns = useMemo(() => {
+    return columns.map((item) => ({
+      ...item,
+      hidden: !checkedList.includes(item.key as string),
+    }));
+  }, [columns, checkedList]);
+
+  const rearrangedColumns = useMemo(
+    () =>
+      newColumns
+        .filter((column) => !column.hidden)
+        .sort((a, b) => {
+          const aIndex = options.findIndex(
+            (option: { value: string; label: string }) => option.value === a.key
+          );
+          const bIndex = options.findIndex(
+            (option: { value: string; label: string }) => option.value === b.key
+          );
+
+          return aIndex - bIndex;
+        }),
+    [newColumns]
+  );
+
+  const handleColumnSelectionDropdownSave = useCallback(() => {
+    setCheckedList(tempCheckedList);
+    setIsDropdownVisible(false);
+  }, [tempCheckedList]);
+
+  const handleColumnSelectionDropdownCancel = useCallback(() => {
+    setTempCheckedList(checkedList);
+    setIsDropdownVisible(false);
+  }, [checkedList]);
+
+  const moveDropdownMenuItem = useCallback(
+    (fromIndex: number, toIndex: number) => {
+      const updatedList = [...options];
+      const [movedItem] = updatedList.splice(fromIndex, 1);
+      updatedList.splice(toIndex, 0, movedItem);
+      setOptions(updatedList);
+      setTempCheckedList((prevCheckedList) => {
+        const updatedCheckedList = prevCheckedList.map((item) => {
+          const index = updatedList.findIndex(
+            (option) => option.value === item
+          );
+
+          return updatedList[index]?.value || item;
+        });
+
+        return updatedCheckedList;
+      });
+    },
+    [options]
+  );
```

Comment:
@shrushti2000 , can we have dropdown-related logic in that component to keep this GlossaryTerm component clean?

Reply:
done